### PR TITLE
move processing of labels and annotations as tags

### PR DESCRIPTION
**Pull request by celenechang**
### What does this PR do?

- Fix issue where a Cluster Agent ClusterRole and ClusterRolebinding was not being cleaned up when DatadogAgent was deleted. This is because of the order of processing of overrides compared to features.
- Changed behavior to only create ClusterRole if it will not be empty
- Also noticed a small typo in one of the conditionals which is now fixed

### Motivation

The presence of the ClusterRole and ClusterRoleBinding was causing a requeue of a nonexistant object

### Additional Notes

Anything else we should know when reviewing?

### Minimum Agent Versions

Are there minimum versions of the Datadog Agent and/or Cluster Agent required?

* Agent: vX.Y.Z
* Cluster Agent: vX.Y.Z

### Describe your test plan

Test according to instructions in https://github.com/DataDog/datadog-operator/pull/1379

In addition:
- Verify that deletion of the DatadogAgent resource results in deletion of the ClusterRole and ClusterRoleBinding called `datadog-cluster-agent-annotations-and-labels-as-tags`
- Verify that removal of both `kubernetesResourcesLabelsAsTags` and `kubernetesResourcesAnnotationsAsTags` from the DatadogAgent manifest results in the absence of the ClusterRole `datadog-cluster-agent-annotations-and-labels-as-tags`.

### Checklist

- [X] PR has at least one valid label: `bug`, `enhancement`, `refactoring`, `documentation`, `tooling`, and/or `dependencies`
- [X] PR has a milestone or the `qa/skip-qa` label

### CODE REVIEW

File: internal/controller/datadogagent/feature/enabledefault/feature.go [hunk 1]
```diff
@@ -101,6 +102,10 @@ type secretInfo struct {
 type clusterAgentConfig struct {
 	serviceAccountName        string
 	serviceAccountAnnotations map[string]string
+
+	resourceMetadataAsTagsClusterRoleName string
+	kubernetesResourcesLabelsAsTags       map[string]map[string]string
+	kubernetesResourcesAnnotationsAsTags  map[string]map[string]string
```
Comment (adel121) [inline internal/controller/datadogagent/feature/enabledefault/feature.go:108]: `kubernetesResourcesLabelsAsTags` and `kubernetesResourcesAnnotationsAsTags` are not specific to cluster agent, it should be available to other components as well (i.e. node agents).

Comment (adel121) [reply internal/controller/datadogagent/feature/enabledefault/feature.go:108]: To elaborate on this:

we previously had, for example, `podLabelsAsTags` which is made available to all components. 

On node agents, this implies that the node agent will use this config to generate tags for pods collected directly from the kubelet.

On CLC runners, this implies that the runner will generate pod tags for unschedulable pods that are collected via informers.

On DCA instance, it means the DCA will collect pods from the API server using workloadmeta and will generate tags based on this config.

So in short, all components need to have access to these 2 configuration values, but only the DCA needs the rbacs

**Review by adel121 — COMMENTED** [review]
Thanks @celenechang , I've left a comment

File: .github/workflows/pr-linter.yml [hunk 2]
```diff
@@ -15,7 +15,7 @@ jobs:
     - name: Check out code into the Go module directory
       uses: actions/checkout@v1
     - name: Verify Pull Request Labels
-      uses: jesusvasquez333/verify-pr-label-action@v1.3.1
+      uses: jesusvasquez333/verify-pr-label-action@v1.4.0
```
[the remark this answers is not in the document]
Comment (celenechang) [reply .github/workflows/pr-linter.yml]: this change was removed from the PR

**Comment by adel121** [comment]
`kubernetesResourcesLabelsAsTags` and `podLabelsAsTags` are very similar in the sense that they should be applied to all 3 agents (node, cluster and runner), and the cluster agent needs an rbac to watch pods. 

I am curious why these 2 features are implemented differently in the operator? 
If we had an rbac issue with `kubernetesResourcesLabelsAsTags` shouldn't we also have the same issue for `podLabelsAsTags`?

**Review by adel121 — COMMENTED** [review]
Question: 

`kubernetesResourcesLabelsAsTags` and `podLabelsAsTags` are very similar in the sense that they should be applied to all 3 agents (node, cluster and runner), and the cluster agent needs an rbac to watch pods.

I am curious why these 2 features are implemented differently in the operator?
If we had an rbac issue with `kubernetesResourcesLabelsAsTags` shouldn't we also have the same issue for `podLabelsAsTags`?

**Review by adel121 — APPROVED** [review]
LGTM

**Review by fanny-jiang — APPROVED** [review]
